cfme/tests/infrastructure/test_vm_migrate.py

```diff
@@ -23,11 +22,19 @@
 def new_vm(setup_provider_modscope, provider, request):
     """Fixture to provision appliance to the provider being tested if necessary"""
     vm_name = random_vm_name(context='migrate')
-    vm = VM.factory(vm_name, provider, template_name=provider.data['small_template'])
+    try:
+        vm = provider.appliance.collections.infra_vms.instantiate(
+            vm_name,
+            provider,
+            provider.data.templates.small_template.name
```

Comment:
I'd move the 'try' to only try getting the template name... to catch AttributeError on exactly this failure, instead of something random that might happen one day when calling `instantiate`:
```python
try:
    template_name = provider.data.templates.small_template.name
except AttributeError:
    ...
```